engine/src/test/java/org/destinationsol/world/generators/SolarSystemGeneratorTest.java

```diff
@@ -0,0 +1,147 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.world.generators;
+
+import org.destinationsol.game.context.Context;
+import org.destinationsol.game.context.internal.ContextImpl;
+import org.destinationsol.modules.ModuleManager;
+import org.destinationsol.world.WorldBuilder;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+
+import static org.destinationsol.world.generators.MazeGenerator.MAZE_BUFFER;
+import static org.destinationsol.world.generators.PlanetGenerator.PLANET_MAX_DIAMETER;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class SolarSystemGeneratorTest {
+    private Context context;
+    private ModuleManager moduleManager;
+    WorldBuilder worldBuilder;
+    SolarSystemGenerator solarSystemGenerator;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        context = new ContextImpl();
+        moduleManager = new ModuleManager();
+        moduleManager.init();
+        context.put(ModuleManager.class, moduleManager);
+        worldBuilder = new WorldBuilder(context, 1);
+
+        ArrayList<SolarSystemGenerator> solarSystemGenerators = worldBuilder.initializeRandomSolarSystemGenerators();
+        solarSystemGenerator = solarSystemGenerators.get(0);
+        setupSolarSystemGenerator();
+    }
+
+    private void setupSolarSystemGenerator() {
+
+        solarSystemGenerator.initializeRandomDefaultFeatureGenerators(1f);
+        solarSystemGenerator.buildFeatureGenerators();
+        solarSystemGenerator.calculateFeaturePositions();
+    }
+
+    @Test
+    void hasCorrectPlanetCountForDefaultSolarSystem() {
+        if(solarSystemGenerator.getClass().equals(SolarSystemGeneratorImpl.class)) {
+            assertEquals(solarSystemGenerator.getPlanetCount(), 5);
+        }
+    }
+
+    @Test
+    void hasCorrectMazeCountForDefaultSolarSystem() {
+        if(solarSystemGenerator.getClass().equals(SolarSystemGeneratorImpl.class)) {
+            assertEquals(solarSystemGenerator.getMazeCount(), 2);
+        }
+    }
+
+    @Test
+    void hasCorrectPossibleBeltCountForDefaultSolarSystem() {
+        if(solarSystemGenerator.getClass().equals(SolarSystemGeneratorImpl.class)) {
+            assertEquals(solarSystemGenerator.getPossibleBeltCount(), 1);
+        }
+    }
+
+    @Test
+    void hasCorrectRadius() {
+        float radiusFromFeatures = 0;
+        radiusFromFeatures += SunGenerator.SUN_RADIUS;
+        for (FeatureGenerator featureGenerator : solarSystemGenerator.getActiveFeatureGenerators()) {
```

Comment:
Shouldn't this only be adding the Orbital radius size?

Reply:
The sun adds radius to the total SolarSystem radius while not being in an orbital (since it is in the center of the system it doesn't orbit anything, but it still has width so it contributes to the radius).

Maybe a good way to think of it is that the first orbital starts at the edge of the sun